### PR TITLE
Remote Datasets: Adjust attributes - remove "title", use default of "name" and "long_name", introduce "description"

### LINKED ISSUE #2914: Remote Datasets: Inconsistent dataset title, name and long_name
A simple script to produce the table below:
```python
from pygmt.datasets.load_remote_dataset import datasets
print("| dataset | title | name | long_name |")
print("|---|---|---|---|")
for name, dataset in datasets.items():
    print("|", " | ".join([name, dataset.title, dataset.name, dataset.long_name]), "|")
```

| dataset | title | name | long_name |
|---|---|---|---|
| earth_age | seafloor age | seafloor_age | age of seafloor crust |
| earth_free_air_anomaly | free air anomaly | free_air_anomaly | IGPP Earth Free-Air Anomaly |
| earth_geoid | Earth geoid | earth_geoid | EGM2008 Earth Geoid |
| earth_magnetic_anomaly | Earth magnetic anomaly | magnetic_anomaly | Earth magnetic anomaly |
| earth_mask | Earth mask | earth_mask | Mask of land and water features |
| earth_relief | Earth relief | elevation | Earth elevation relative to the geoid |
| earth_vgg | Earth vertical gravity gradient | earth_vgg | IGPP Earth Vertical Gravity Gradient |
| earth_wdmam | WDMAM magnetic anomaly | wdmam | World Digital Magnetic Anomaly Map |

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Currently the naming of the remote datasets is inconistent (please see https://github.com/GenericMappingTools/pygmt/issues/2914#issue-2056175156). This PR applies the following changes to improve consistency:

- [x] Keep all the `load_*` API functions unchanged
- [x] Make dataset match the GMT remote dataset name, e.g., `earth_age`
- [x] Remove dataset attribute `title`
- [x] Do not set dataset attribute `name` explicitly anymore, as it defaults now to ``"z"``
       (see #3120)
- [x] Do not set dataset attribute `long_name` explicitly anymore, as it defaults now to `"quantity (unit)"`
       (see #3120)
- [x] Introduce dataset attribute `description`. It should match the title on the dataset documentation page (https://www.generic-mapping-tools.org/remote-datasets/), but use lower-case letter when possible e.g., `EarthByte Earth seafloor crustal age`, rather than `EarthByte Earth Seafloor Crustal Age`.
- [x] Split `earth_relief` into `earth_gebco` and `earth_igpp` in `load_remote_dataset.py`
- [x] Sort datasets alphetically in `load_remote_dataset.py`

Fixes #2914 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

File: pygmt/datasets/earth_vertical_gravity_gradient.py
Comment (seisman) [inline pygmt/datasets/earth_vertical_gravity_gradient.py:2]:
```suggestion
Function to download the IGPP Earth vertical gravity gradient dataset from the GMT data
```

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:159]: Perhaps we should have separate entries for `earth_relief` and `earth_gebco` since they're two different datasets.

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:83]:
```suggestion
    "earth_mag": GMTRemoteDataset(
        name="earth_mag",
```

File: pygmt/datasets/earth_magnetic_anomaly.py
Comment (seisman) [inline pygmt/datasets/earth_magnetic_anomaly.py:149]:
```suggestion
    dataset_name = "earth_wdmam" if data_source == "wdmam" else "earth_mag"
```

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:44]: Prefer to remove it.
```suggestion
        The name assigned as an attribute to the DataArray.
```

File: pygmt/datasets/load_remote_dataset.py
Comment (yvonnefroehlich) [reply pygmt/datasets/load_remote_dataset.py]: This is done in the commits https://github.com/GenericMappingTools/pygmt/pull/3048/commits/15f42b72064deba77870b627d93a17c98a879996 - https://github.com/GenericMappingTools/pygmt/pull/3048/commits/3ca6f323f11545695b8a54b59cc3f80fbee94b98

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:85]: https://www.generic-mapping-tools.org/remote-datasets/earth-mag.html
```suggestion
        long_name="EMAG2 Earth magnetic anomaly model",
```

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:83]: Please also move the `earth_mag` entry below the `earth_mask` entry.

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:227]: https://www.generic-mapping-tools.org/remote-datasets/earth-wdmam.html
```suggestion
        long_name="WDMAM world digital magnetic anomaly map",
```

File: pygmt/tests/test_datasets_earth_magnetic_anomaly.py
Comment (yvonnefroehlich) [inline pygmt/tests/test_datasets_earth_magnetic_anomaly.py:18]:
```suggestion
    assert data.attrs["long_name"] == "EMAG2 Earth Magnetic Anomaly Model"
```

File: pygmt/tests/test_datasets_earth_magnetic_anomaly.py
Comment (yvonnefroehlich) [inline pygmt/tests/test_datasets_earth_magnetic_anomaly.py:64]:
```suggestion
    assert data.attrs["long_name"] == "EMAG2 Erth Magnetic Anomaly Model"
```

File: pygmt/tests/test_datasets_earth_magnetic_anomaly.py
Comment (yvonnefroehlich) [inline pygmt/tests/test_datasets_earth_magnetic_anomaly.py:120]:
```suggestion
    assert data.attrs["long_name"] == "WDMAM World Digital Magnetic Anomaly Map"
```

File: pygmt/datasets/load_remote_dataset.py
Comment (yvonnefroehlich) [reply pygmt/datasets/load_remote_dataset.py]: The alphabetical order was updated in https://github.com/GenericMappingTools/pygmt/pull/3048/commits/ab00c939bcb6ae21b972a52efefea6337e33d5d6.

File: examples/tutorials/advanced/grid_equalization.py
Comment (seisman) [inline examples/tutorials/advanced/grid_equalization.py:23]: I'm a little unsure if it's a good idea to use the `dataset.name` as the `grd2xyz` output column name. It's seems a little weird to have a column named `earth_igpp`.

Comment (yvonnefroehlich) [reply examples/tutorials/advanced/grid_equalization.py:23]: Yes, I was also not so happy with this when changing it. I personally prefer column names that contain the actual quantity (and in the best case, the unit), i.e., here `elevation` (`elevation_m`).

Comment (seisman) [reply examples/tutorials/advanced/grid_equalization.py:23]: @yvonnefroehlich Sorry for the slow response. I think we probably need to finalize PR  https://github.com/GenericMappingTools/pygmt/pull/2398 first because changes #2398 may affect our decisions.

Comment (yvonnefroehlich) [reply examples/tutorials/advanced/grid_equalization.py:23]: No worries. Actually, I think your response is here https://github.com/GenericMappingTools/pygmt/issues/2914#issuecomment-1999167297.
We can simply leave this PR as is and wait until #2398 is merged.

Comment (yvonnefroehlich) [reply examples/tutorials/advanced/grid_equalization.py:23]: Looks like we can try to continue here, as PR #2398 has been merged.

Comment (seisman) [reply examples/tutorials/advanced/grid_equalization.py:23]: Sorry, still need to wait for #3120.

Comment (yvonnefroehlich) [reply examples/tutorials/advanced/grid_equalization.py:23]: Yes sure, no worries. Try the find time to look at your comments in PR #2914.

File: pygmt/datasets/load_remote_dataset.py
Comment (seisman) [inline pygmt/datasets/load_remote_dataset.py:69]: With PR #3120 merged, the grid name defaults to `z` and `long_name` defaults to `ages (Myr)`. So we can remove the `name` and `long_name` attributes.
```suggestion
```

File: pygmt/datasets/earth_relief.py
Comment (seisman) [inline pygmt/datasets/earth_relief.py:168]: Rewrite use match-case statements:
```suggestion
    match data_source:
        case "igpp" | "synbath":
            dataset_name = "earth_igpp"
        case "gebco" | "gebcosi":
            dataset_name = "earth_gebco"
```

File: examples/tutorials/advanced/grid_equalization.py
Comment (seisman) [inline examples/tutorials/advanced/grid_equalization.py:23]: Also need to update https://github.com/GenericMappingTools/pygmt/blob/8b2a74ca8e68c1d7e34e46b089bf522e7ba683f3/pygmt/src/grd2xyz.py#L142.

**Review by seisman — APPROVED** [review]
Looks good!

**Comment by seisman** [comment]
Need to update the PR title to reflect the changes.

**Comment by yvonnefroehlich** [comment]
> Need to update the PR title to reflect the changes.

I update the description of this PR in the initial comment at the beginning and tried to find a better title (not 100 % happy with it - feel free to adjust).

**Comment by yvonnefroehlich** [comment]
The failing `macos-latest - Python 3.10 / NumPy 1.23 (pull_request)` and `macos-latest - Python 3.12 / NumPy 1.26 (pull_request)` tests are probably related to issue #3194.

But there is also a failing windows test `windows-latest - Python 3.10 / NumPy 1.23 (pull_request)`. However, `windows-latest - Python 3.12 / NumPy 1.26 (pull_request)` is successful.

**Comment by seisman** [comment]
> The failing `macos-latest - Python 3.10 / NumPy 1.23 (pull_request)` and `macos-latest - Python 3.12 / NumPy 1.26 (pull_request)` tests are probably related to issue #3194.
> 
> But there is also a failing windows test `windows-latest - Python 3.10 / NumPy 1.23 (pull_request)`. However, `windows-latest - Python 3.12 / NumPy 1.26 (pull_request)` is successful.

The Windows failure is unrelated to this PR.

**Comment by seisman** [comment]
@yvonnefroehlich Nice work. Thanks!

**Pull request merged**
